src/swerex/runtime/remote.py

```diff
@@ -164,16 +167,48 @@ async def is_alive(self, *, timeout: float | None = None) -> IsAliveResponse:
     async def wait_until_alive(self, *, timeout: float = 60.0):
         return await _wait_until_alive(self.is_alive, timeout=timeout)
 
-    async def _request(self, endpoint: str, request: BaseModel | None, output_class: Any):
+    async def _request(self, endpoint: str, request: BaseModel | None, output_class: Any, num_retries: int = 10):
         """Small helper to make requests to the server and handle errors and output."""
-        session = await self._ensure_session()
+        request_url = f"{self._api_url}/{endpoint}"
+        return await self._request_with_retry(request_url, request, output_class, num_retries)
 
-        async with session.post(
-            f"{self._api_url}/{endpoint}", json=request.model_dump() if request else None, headers=self._headers
-        ) as response:
-            await self._handle_response_errors(response)
-            data = await response.json()
-            return output_class(**data)
+    async def _request_with_retry(
+        self,
+        request_url: str,
+        request: BaseModel | None,
+        output_class: Any,
+        num_retries: int,
+    ):
+        """Small helper to make requests to the server and handle errors and output."""
+        request_id = str(uuid.uuid4())
+        headers = self._headers.copy()
+        headers["X-Request-ID"] = request_id
+
+        retry_count = 0
+        last_exception = None
+        retry_delay = 0.1
+        backoff_max = 5
+
+        async with aiohttp.ClientSession() as session:
+            while retry_count <= num_retries:
```

Comment:
Nice, so this works when you inject the error?

Reply:
Yep! This works at a 70% error rate (90% success with 3 retries)